index.rst

```diff
@@ -48,12 +48,336 @@
+
+In cases where one or more private ``RUN`` collections contain datasets that should _not_ be considered part of the final public outputs (e.g. because they are superceded by datasets in other private ``RUN`` collection(s), a ``TAGGED`` collection can be used to screen and aggregate these.
+That ``TAGGED`` collection would then be a direct child of the final public ``CHAINED`` collection, instead of any ``RUN`` collections it references.
+
+.. note::
+
+   It is not generally possible to use a ``TAGGED`` collection as the public output collection for a processing run, because putting master calibrations (which are almost always inputs, even if indirectly) in a ``TAGGED`` collection strips them of their validity ranges and does not allow datasets from different validity ranges to coexist.
+   So even if a ``TAGGED`` collection is used, the public ``CHAINED`` collection would contain both that collection and the input ``CALIBRATION`` collection as children.
+
+.. note::
+
+   These public ``CHAINED`` collections essentially mimic Gen2's "parent link" mechanism, which provides at best approximate coarse-grained provenance information about which datasets were used as inputs when producing others.
+   The Gen3 repository will eventually be extended to include fine-grained, exact provenance - essentially a serialization of the directed acyclic graph (DAG) that describes the processing).
+   Whether queries against that DAG are fast enough to allow this more rigorous provenance information to be used as a type of collection (replacing some usage of ``TAGGED`` and ``CHAINED`` collections) remains to be seen, however.
+   It is also worth noting that in general the full DAG does not maintain the usual collection invariant of having only one dataset with a particular dataset type and data ID (e.g. two calexps with the same data ID, from two differently-configured runs, could each contribute to different coadd patches in a downstream run).
+
+.. _collections-developer-processing-outputs:
+
+Developer processing outputs
+----------------------------
+
+Processing initiated by DM developers that are intented primarily for personal or small-group use must start with ``u/<user>`` (e.g. ``u/jbosch``), and are strongly encouraged to start with ``u/<user>/<ticket>`` (e.g. ``u/jbosch/DM-56789``) whenever possible.
+Names and structure after this prefix are at user discretion, but we strongly recommend using a combination of ``CHAINED`` collections and ``RUN`` collections to distinguish between "inputs and outputs" collections and "output only" collections, as in :ref:`collections-shared-official-processing-outputs`.
+The ``pipetask`` tool will automatically take care of this if the ``--output`` option is used with or instead of the ``--output-run`` option.
+
+.. note::
+   **TODO**: does BPS provide similar functionality?
```

Comment:
We would need to double check details.  BPS allows users to specify whatever output collection they want (example yaml shows "shared/ci_hsc_gen3/{timestamp}" but could easily be changed to the u pattern).  It calls pipetask with --output-run  for the --init-only execution as well as each quantum execution.  I think this came from wanting to leave the ability for production services to use other terms for uniqueness than timestamp if it wanted to (e.g., processing attempt number).